### PR TITLE
add trilogy support

### PR DESCRIPTION
**Pull request by joevandyk**
adds suport for the trilogy mysql-compatible adapter

fixes https://github.com/ClosureTree/with_advisory_lock/issues/78

we've been using this change in production for a couple weeks, no issues noted.

### CODE REVIEW

**Comment by skipkayhil** [comment]
@seuros would you mind taking a look at this? With Rails 7.1 out, there's going to be an increase in demand for using Trilogy

File: .tool-versions [hunk 1]
```diff
@@ -1 +1 @@
-ruby 3.0.5
+ruby 3.2.1
```
Comment (joevandyk) [inline .tool-versions:1]: (i can revert this if needed)

Comment (seuros) [reply .tool-versions:1]: It fine. Btw 3.2.2 is latest

**Comment by seuros** [comment]
Did you check my open pr ? Seem trilogy is failling in some test.

I will take a look again

File: gemfiles/activerecord_6.0.gemfile [hunk 2]
```diff
@@ -8,6 +8,7 @@ platforms :ruby do
   gem "sqlite3"
   gem "mysql2"
   gem "pg"
+  gem "activerecord-trilogy-adapter
```
Comment (joevandyk) [inline gemfiles/activerecord_6.0.gemfile:11]:
```suggestion
  gem "activerecord-trilogy-adapter "
```

File: gemfiles/activerecord_6.0.gemfile [hunk 3]
```diff
@@ -8,6 +8,7 @@ platforms :ruby do
   gem "sqlite3"
   gem "mysql2"
   gem "pg"
+  gem "activerecord-trilogy-adapter "
```
Comment (joevandyk) [inline gemfiles/activerecord_6.0.gemfile:11]:
```suggestion
  gem "activerecord-trilogy-adapter"
```

File: gemfiles/activerecord_6.1.gemfile [hunk 4]
```diff
@@ -8,6 +8,7 @@ platforms :ruby do
   gem "sqlite3"
   gem "mysql2"
   gem "pg"
+  gem "activerecord-trilogy-adapter
```
Comment (joevandyk) [inline gemfiles/activerecord_6.1.gemfile:11]:
```suggestion
  gem "activerecord-trilogy-adapter"
```

File: gemfiles/activerecord_7.0.gemfile [hunk 5]
```diff
@@ -8,6 +8,7 @@ platforms :ruby do
   gem "sqlite3"
   gem "mysql2"
   gem "pg"
+  gem "activerecord-trilogy-adapter
```
Comment (joevandyk) [inline gemfiles/activerecord_7.0.gemfile:11]:
```suggestion
  gem "activerecord-trilogy-adapter"
```

File: gemfiles/activerecord_edge.gemfile [hunk 6]
```diff
@@ -8,6 +8,7 @@ platforms :ruby do
   gem "sqlite3"
   gem "mysql2"
   gem "pg"
+  gem "activerecord-trilogy-adapter
```
Comment (joevandyk) [inline gemfiles/activerecord_edge.gemfile:11]:
```suggestion
  gem "activerecord-trilogy-adapter"
```

**Comment by skipkayhil** [comment]
> Did you check my open pr ? Seem trilogy is failling in some test.

Ah, I didn't see that one! I'll check it out

**Comment by seuros** [comment]
Thank you for your PR. 

Already supported in latest version, including legacy versions.

Let me know if you face any error.

**Pull request closed without merging**